pkg/render/intrusion_detection.go

```diff
@@ -1404,18 +1409,27 @@ func (c *intrusionDetectionComponent) adPersistentVolumeClaim() *corev1.Persiste
 
 func (c *intrusionDetectionComponent) adAPIDeployment(configureADStorage bool) *appsv1.Deployment {
 	var adModelVolumeSource corev1.VolumeSource
+	sc := securitycontext.NewBaseContext(securitycontext.RunAsUserID, securitycontext.RunAsGroupID)
+
 	if configureADStorage {
 		adStorageClassName := c.cfg.IntrusionDetection.Spec.AnomalyDetection.StorageClassName
 		adModelVolumeSource = corev1.VolumeSource{
 			PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
 				ClaimName: adStorageClassName,
 			},
 		}
+
+		// When Persistent Storage is configured for AD API, it will use the host path of /var/anomaly-detection-api
+		// where it requires root privileges to write to
+		sc.RunAsGroup = ptr.Int64ToPtr(0)
+		sc.RunAsUser = ptr.Int64ToPtr(0)
+		sc.Privileged = ptr.BoolToPtr(true)
```

Comment:
Do you want to set `Provileged` to be openshift only like here. otherwise, leave it non-privileged?

https://github.com/tigera/operator/blob/c0be4fb5cc3535b3404a21b7d2895ea9cfef3b43/pkg/render/intrusion_detection.go#L642

Reply:
just going to set the privileged to `false` https://github.com/tigera/operator/pull/2300/commits/19a50c7bc33cd00d8728c3eba55b27501675bb0e - tested in openshift doesnt need to be priivileged